front/pages/api/data_sources/[user]/[name]/search/index.ts

```diff
@@ -0,0 +1,193 @@
+import { Op } from "sequelize";
+import { credentialsFromProviders } from "@app/lib/providers";
+import {
+  parse_payload,
+  RequestParseError,
+  json_error,
+} from "@app/lib/http_utils";
+import { JSONSchemaType } from "ajv";
+
+interface TagsFilter {
+  is_in?: string[];
+  is_not?: string[];
+}
+
+interface TimestampFilter {
+  gt?: number;
+  lt?: number;
+}
+
+interface SearchFilter {
+  tags?: TagsFilter;
+  timestamp?: TimestampFilter;
+}
+
+interface DatasourceSearchQuery {
+  query: string;
+  top_k: number;
+  filter?: SearchFilter;
+  full_text: boolean;
+}
+
+const search_query_schema: JSONSchemaType<DatasourceSearchQuery> = {
+  type: "object",
+  properties: {
+    query: { type: "string" },
+    top_k: { type: "number" },
+    full_text: { type: "boolean" },
+    filter: {
+      type: "object",
+      nullable: true,
+      properties: {
+        tags: {
+          type: "object",
+          nullable: true,
+          properties: {
+            is_in: { type: "array", items: { type: "string" }, nullable: true },
+            is_not: {
+              type: "array",
+              items: { type: "string" },
+              nullable: true,
+            },
+          },
+        },
+        timestamp: {
+          type: "object",
+          nullable: true,
+          properties: {
+            gt: { type: "number", nullable: true },
+            lt: { type: "number", nullable: true },
+          },
+        },
+      },
+    },
+  },
+  required: ["query", "top_k", "full_text"],
+};
+
+const { DUST_API } = process.env;
+
+/**
+ *
+ * @param req
+ * @param res
+ */
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const session = await getServerSession(req, res, authOptions);
+
+  let user = await User.findOne({
+    where: {
+      username: req.query.user,
+    },
+  });
+
+  if (!user) {
+    res.status(404).end();
+    return;
+  }
+
+  const readOnly = !(
+    // @ts-expect-error
+    (session && session.provider.id.toString() === user.githubId)
+  );
+
+  let dataSource = await DataSource.findOne({
+    where: readOnly
+      ? {
+          // @ts-expect-error
+          userId: user.id,
+          name: req.query.name,
+          visibility: {
+            [Op.or]: ["public"],
+          },
+        }
+      : {
+          // @ts-expect-error models are not type script friendly yet
+          userId: user.id,
+          name: req.query.name,
+        },
+    attributes: [
+      "id",
+      "name",
+      "description",
+      "visibility",
+      "config",
+      "dustAPIProjectId",
+      "updatedAt",
+    ],
+  });
+
+  if (!dataSource) {
+    res.status(404).end();
+    return;
+  }
+
+  switch (req.method) {
+    case "POST":
+      let search_query: DatasourceSearchQuery | null = null;
+      try {
+        search_query = parse_payload(search_query_schema, req.body);
+        console.log("successfull parse", JSON.stringify(search_query));
```

Comment:
Let's do
```
console.log(`Received query: path=${query path} body=${JSON.stringify(search_query)}`
```

It was super useful at Stripe to have a uniform way to log stuff. I propose `static message: var1=... var2=...`

Reply:
I suggest we take care of that in another PR. This was just a left over from my debugging messages.